### PR TITLE
Update HACKING.rst to have more information

### PR DESCRIPTION
**Pull request by kylehogan**
I just did an update to HACKING.rst with some more specifics on how to get haas set up. One more thing I would like to add is how to configure the test environment to use PostgreSQL

### CODE REVIEW

File: HACKING.rst [hunk 1]
```diff
@@ -1,44 +1,52 @@
...
+available as RPMs as well, but we recommend installing them with pip, since
+this will install the versions that HaaS has been tested with.  This is done
+automatically by the instructions below.
 
-  AttributeError: 'NoneType' object has no attribute 'skip_requirements_regex'
+Creating HaaS System User
+=========================
 
-Fix this by upgrading ``pip`` within the virtual environment::
+Create a system user ``haas_user`` with::
```
Comment (SahilTikale) [inline HACKING.rst]: I always wanted the user to be simply `haas` and not `haas_user`.
If you have no objection against it, I suggest changing it here.

File: HACKING.rst [hunk 2]
```diff
@@ -1,44 +1,52 @@
...
 
-  pip install --upgrade pip
+  sudo useradd haas_user -d /var/lib/haas -m -r
```
Comment (SahilTikale) [inline HACKING.rst]: This is the reason I wanted the user to be `haas` and not `haas_user`
It has been always a good practice to have the home directory and username to be the same string. 
Helps manage the system better.

File: HACKING.rst [hunk 3]
```diff
@@ -55,7 +63,58 @@ server without requiring physical hardware. To suppress actual node and headnode
...
+  sudo chown haas_user:haas_user /etc/haas.cfg
+  sudo chmod 400 /etc/haas.cfg
+
+Setting up Database
+===================
+
+If using SQLite no additional setup is necessary. Should ensure that `uri` in `database` setion of 'haas.cfg' is set to `sqlite:///haas.db` and can skip to initializing the database.
+
+If using PostgreSQL, follow these steps to setup the database:
+
```
Comment (SahilTikale) [inline HACKING.rst]: At this point simply point the user to my document #592.

**Comment by SahilTikale** [comment]
@kylehogan #592 restructures the Installation related docs for both development and production.
Could you take a look at that and help to merge it before we merge this one ?

File: HACKING.rst [hunk 4]
```diff
@@ -55,7 +63,58 @@ server without requiring physical hardware. To suppress actual node and headnode
...
+  createdb haas_db -O haas_user
+  
+setup haas.cfg to access Postgres database:
+
+the ``uri`` option in haas.cfg must be set to::
+
+   postgresql://<user>:<password>@<address>/<dbname>
+
+Where ``<user>`` is the name of the postgres user you created (here this is ``haas_user``), ``<password>`` is its password, ``<dbname>`` is the name of the database you created (here this is ``haas_db``), and ``<address>`` is the address which HaaS should use to connect to postgres (In a typical default postgres setup, the right value is ``localhost``).
+
```
Comment (SahilTikale) [inline HACKING.rst]: delete all the lines including 81 to 110 since the information will get replaced by document added by #592

**Comment by shwsun** [comment]
Some doc @kylehogan added was what I meant today, the HACKING.rst should provide enough info for a developer to start, further info like install postgres on centos, or howto fix a error with Debian Wheezy and Ubuntu 12.04 should goto some FAQ.rst.

@SahilTikale if you agree we can merge some doc into #592 and start from there?

**Comment by matsuuran** [comment]
@shwsun @kylehogan @SahilTikale did this get rolled into #592?

**Comment by SahilTikale** [comment]
Not yet. Was working a paper. Will sort it out tomorrow.

**Comment by SahilTikale** [comment]
I have included these instructions in PR #592 
This can be closed.

**Pull request closed without merging**